pyk/src/pyk/prelude/collections.py

```diff
@@ -51,3 +52,16 @@ def map_item(k: KInner, v: KInner) -> KInner:
 def map_of(ks: dict[KInner, KInner] | Iterable[tuple[KInner, KInner]]) -> KInner:
     ks = dict(ks)
     return build_assoc(map_empty(), KLabel('_Map_'), (map_item(k, v) for k, v in ks.items()))
+
+
+def rangemap_empty() -> KInner:
+    return KApply('.RangeMap')
+
+
+def rangemap_item(k: tuple[KInner, KInner], v: KInner) -> KInner:
+    return KApply('_r|->_', [KApply('RangeMap:Range', k), v])
+
+
+def rangemap_of(ks: dict[tuple[KInner, KInner], KInner] | Iterable[tuple[tuple[KInner, KInner], KInner]]) -> KInner:
+    ks_dict: dict[tuple[KInner, KInner], KInner] = dict(ks)
```

Comment:
Does the type checker complain for this?

```suggestion
    ks_dict = dict(ks)
```

Reply:
Without the annotation, it infers the type as `dict[object, KInner]` and reports an error on the `rangemap_item(k, v)` call below.